File: app2/src/main/java/guoan/app2/MainActivity.java
package guoan.app2;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.TextView;

import com.guoan.community.sdk.CommunityFactory;

/**
 * Created by andylove on 2018/3/1.
 */

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        TextView enter = (TextView) findViewById(R.id.cl_enter_sdk);
        enter.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                CommunityFactory.Companion.getInstance().onIntoCommunityHome(MainActivity.this);
            }
        });

    }

}

File: app/src/main/java/com/guoan/sdkdemo/MainActivity.kt
package com.guoan.sdkdemo

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.guoan.community.sdk.CommunityConstants
import com.guoan.community.sdk.CommunityFactory
import com.guoan.community.sdk.userinfo.CommunityLoginBack
import kotlinx.android.synthetic.main.activity_main.*
import org.jetbrains.anko.onClick
import org.jetbrains.anko.toast

/**
 * @author andylove
 * demo pager
 */
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        ck_enter_sdk?.onClick {
            CommunityFactory.getInstance()?.onIntoCommunityHome(this@MainActivity, null, true, null)
        }
        login?.onClick {
            startActivity(Intent(this@MainActivity, LoginActivity::class.java))
        }
        loc?.onClick {
            startActivity(Intent(this@MainActivity, LocationActivity::class.java))
        }
        exit?.onClick {
            finish()
            CommunityFactory.onDestory()
        }
        community_login.onClick {
            CommunityFactory.getInstance()?.onLoginCommunity(this@MainActivity)
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (CommunityConstants.SDK_COMMUNITY_LOGION_CODE == requestCode && data != null) {
            var loginBack: CommunityLoginBack? = data?.getSerializableExtra(CommunityConstants.SDK_COMMUNITY_LOGION_KEY) as CommunityLoginBack?
            toast("登录成功 token：${loginBack?.customer?.appToken}")
        }
    }

}
